src/System.Net.Http/src/System/Net/Http/HttpRuleParser.cs

```diff
@@ -394,43 +392,38 @@ internal static HttpParseResult GetQuotedPairLength(string input, int startIndex
-                            return HttpParseResult.InvalidFormat;
-                        }
+                        return HttpParseResult.InvalidFormat;
+                    }
 
-                        int nestedLength = 0;
-                        HttpParseResult nestedResult = GetExpressionLength(input, current, openChar, closeChar,
-                            supportsNesting, ref nestedCount, out nestedLength);
+                    int nestedLength = 0;
+                    HttpParseResult nestedResult = GetExpressionLength(input, current, openChar, closeChar,
+                        supportsNesting, nestedCount, out nestedLength);
 
-                        switch (nestedResult)
-                        {
-                            case HttpParseResult.Parsed:
-                                current += nestedLength; // Add the length of the nested expression and continue.
-                                break;
-
-                            case HttpParseResult.NotParsed:
-                                Debug.Fail("'NotParsed' is unexpected: We started nested expression " +
-                                    "parsing, because we found the open-char. So either it's a valid nested " +
-                                    "expression or it has invalid format.");
-                                break;
-
-                            case HttpParseResult.InvalidFormat:
-                                // If the nested expression is invalid, we can't continue, so we fail with invalid format.
-                                return HttpParseResult.InvalidFormat;
-
-                            default:
-                                Debug.Fail("Unknown enum result: " + nestedResult);
-                                break;
-                        }
-                    }
-                    finally
+                    switch (nestedResult)
                     {
-                        nestedCount--;
+                        case HttpParseResult.Parsed:
+                            current += nestedLength; // Add the length of the nested expression and continue.
+                            break;
+
+                        case HttpParseResult.NotParsed:
+                            Debug.Fail("'NotParsed' is unexpected: We started nested expression " +
+                                "parsing, because we found the open-char. So either it's a valid nested " +
+                                "expression or it has invalid format.");
+                            break;
+
+                        case HttpParseResult.InvalidFormat:
+                            // If the nested expression is invalid, we can't continue, so we fail with invalid format.
+                            return HttpParseResult.InvalidFormat;
+
+                        default:
+                            Debug.Fail("Unknown enum result: " + nestedResult);
+                            break;
                     }
+                    nestedCount--;
```

Comment:
We can remove try/finally now lifecycle of `nestedCount` is current stackframe so we only need to keep correct se we decrement if inner parsing result switch `break` in case of invalid we return `InvalidFormat` to caller(until end of recursion) so it's unuseful keep "current frame" `nestedCount`.